modules/gapi/test/infer/gapi_infer_onnx_test.cpp

```diff
@@ -442,32 +548,92 @@ TEST_F(ONNXGRayScaleTest, InferImage)
+    infer<uint8_t>(prep_mat, out_onnx);
+    cv::Mat onnx_conv_out({1, 1, 200, 7}, CV_32F);
+    remapToIESsdOut({out_onnx[3], out_onnx[0], out_onnx[2], out_onnx[1]}, onnx_conv_out);
+    out_onnx.clear();
+    out_onnx.push_back(onnx_conv_out);
     // G_API code
-    using SSDOut = std::tuple<cv::GMat, cv::GMat, cv::GMat, cv::GMat>;
-    G_API_NET(MobileNet, <SSDOut(cv::GMat)>, "ssd_mobilenet");
+    G_API_NET(MobileNet, <cv::GMat(cv::GMat)>, "ssd_mobilenet");
     cv::GMat in;
-    cv::GMat out1, out2, out3, out4;
-    std::tie(out1, out2, out3, out4) = cv::gapi::infer<MobileNet>(in);
-    cv::GComputation comp(cv::GIn(in), cv::GOut(out1, out2, out3, out4));
-    auto net = cv::gapi::onnx::Params<MobileNet>{model_path}
-        .cfgOutputLayers({"out1", "out2", "out3", "out4"})
-        .cfgPostProc({cv::GMatDesc{CV_32F, {1, 100, 4}},
-                      cv::GMatDesc{CV_32F, {1, 100}},
-                      cv::GMatDesc{CV_32F, {1, 100}},
-                      cv::GMatDesc{CV_32F, {1, 1}}}, remapSsdPorts);
-    out_gapi.resize(num_out);
+    cv::GMat out = cv::gapi::infer<MobileNet>(in);
+    cv::GComputation comp(cv::GIn(in), cv::GOut(out));
+    auto net = cv::gapi::onnx::Params<MobileNet>{ model_path }
+        .cfgOutputLayers({"detection_output"})
+        .cfgPostProc({cv::GMatDesc{CV_32F, {1, 1, 200, 7}}}, remapSsdPorts);
     comp.apply(cv::gin(in_mat1),
-               cv::gout(out_gapi[0], out_gapi[1], out_gapi[2], out_gapi[3]),
+               cv::gout(out_gapi.front()),
                cv::compile_args(cv::gapi::networks(net)));
     // Validate
     validate();
 }
+
+TEST_F(ONNXYoloV3MultInput, InferConstInput)
+{
+    useModel("/object_detection_segmentation/yolov3/model/yolov3-10");
+    // ONNX_API code
+    infer<float>(ins, out_onnx);
+    // G_API code
+    using OUT = std::tuple<cv::GMat, cv::GMat, cv::GMat>;
+    G_API_NET(YoloNet, <OUT(cv::GMat)>, "yolov3");
+    auto net = cv::gapi::onnx::Params<YoloNet>{model_path}
+        .constInput("image_shape", ins[1])
+        .cfgInputLayers({"input_1"})
+        .cfgOutputLayers({"out1", "out2", "out3"})
+        .cfgPostProc({cv::GMatDesc{CV_32F, {1, 10000, 4}},
+                      cv::GMatDesc{CV_32F, {1, 80, 10000}},
+                      cv::GMatDesc{CV_32S, {5, 3}}}, remapYoloV3);
+    cv::GMat in, out1, out2, out3;
+    std::tie(out1, out2, out3) = cv::gapi::infer<YoloNet>(in);
+    cv::GComputation comp(cv::GIn(in), cv::GOut(out1, out2, out3));
+    out_gapi.resize(num_out);
+    comp.apply(cv::gin(ins[0]),
+               cv::gout(out_gapi[0], out_gapi[1], out_gapi[2]),
+               cv::compile_args(cv::gapi::networks(net)));
+    // Validate
+    validate();
+}
+
+TEST_F(ONNXYoloV3MultInput, InferBSConstInput)
+{
+    useModel("/object_detection_segmentation/yolov3/model/yolov3-10");
+    // Tensor with incorrect image size
+    // is used for check case when InputLayers and constInput have same names
+    cv::Mat bad_shape;
```

Comment:
Is it a bad arg test? Which check should fail?

Reply:
This test checks use of data from const input when input names are incorrectly set. If input names contain names from const input, then calculation will be performed according to data given by `.ConstInput()` function.